### PR TITLE
T validationworkflow

### PR DESCRIPTION
**Pull request by jakebrown58**
Changes the UI for the deployment job submission page to make it into a 2-page wizard with validations.

### CODE REVIEW

File: src/server/controllers/envs.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ exports.list = function list(req, res, next) {
 
 exports.get = function get(req, res, next) {
   var envId = req.params.envId;
-  debug('getting environment - findById %s', req.params.envId);
+  //debug('getting environment - findById %s', req.params.envId);
```
Comment (bmancini55) [inline src/server/controllers/envs.js]: Left comment in here...

File: src/client/app/validations/details.controller.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+(function () {
+  'use strict';
+
+  angular
+    .module('app')
+    .controller('ValidationDetailsController', ValidationDetailsController);
```
Comment (bmancini55) [inline src/client/app/validations/details.controller.js]: One pattern this app has is only using controllers for route handling.  This decouples the routing from the components.  This _could_ be refactored into `details-route.controller.js` and `details-route.html` which loads `details.directive.js` and `details.html` with the actual logic here.

File: src/server/controllers/envs.js (hunk 1 above)
Comment (bmancini55) [inline src/server/controllers/envs.js]: Comment?

File: src/server/services/env-service.js [hunk 3]
```diff
@@ -286,7 +295,7 @@ function joinEnvSkytap(env) {
         env.runstate = skyenv.runstate;
         return env;
       }, function() {
-        debug('Could not find env %d', env.remoteId);
+        //debug('Could not find env %d', env.remoteId);
```
Comment (bmancini55) [inline src/server/services/env-service.js]: Comment?

File: src/server/controllers/jobs.js [hunk 4]
```diff
@@ -2,23 +2,99 @@ let debug       = require('debug')('deployer-projects');
...
+  debug('getValidations ' + jobId);
+  validationrunner.getValidation(jobId, (err, data) => {
+    let runLogs = [];
+
+    // data.tasks structure looks like this:
+    //    data: { tasks: [{ subtask: {name: '', runLog: ['message']}]}
+    // we're trying to flatten it here like this:
+    //    runLogs: [message: 'message', machine: subtask.name] }
+
+    _.each(data.tasks, function(task) {
```
Comment (adampaulberg) [inline src/server/controllers/jobs.js]: alternatively you can do this without underscore

```
for(let task of data.tasks)
```

File: src/server/controllers/jobs.js [hunk 5]
```diff
@@ -45,6 +121,141 @@ exports.summaryList = function summaryList(req, res, next) {
...
+
+    let recentFailures = _.filter(recent, function(item) {
+      return item.status === 'Failed';
+    });
+
+    let recentSuccesses = _.filter(recent, function(item) {
+      return item.status.startsWith('Suc');
+    });
+
+    // let nonAllinone = _.filter(recentFailures, function(item) {
```
Comment (adampaulberg) [inline src/server/controllers/jobs.js]: comment?

File: src/server/controllers/jobs.js [hunk 6]
```diff
@@ -45,6 +121,141 @@ exports.summaryList = function summaryList(req, res, next) {
...
+    //         isAllInOne = item.text.tasks[0].tasks[0].name.toLowerCase().startsWith('all');
+    //     }
+    //     catch (e) {
+    //       isAllInOne = false;
+    //     }
+    //   return isAllInOne;
+    // });
+    let outcomesByEnv = _.mapObject(_.groupBy(recent, 'name'), function(val) {
+       return _.map(val, function(v) {
+          return 'http://localhost:11234/app/jobs/' + v.id;
```
Comment (adampaulberg) [inline src/server/controllers/jobs.js]: should this port come from a config?

File: src/client/app/validations/details.controller.js (hunk 2 above)
Comment (jakebrown58) [inline src/client/app/validations/details.controller.js]: ok - good feedback.

i was copying the job page.  (copy / pasting).

i spent more time than I wanted trying to get the directive pattern working, and i couldn't.... so i punted for a future refactor.  this page isn't actually used in the job-sumission workflow anyway.

File: src/server/controllers/jobs.js (hunk 6 above)
Comment (jakebrown58) [inline src/server/controllers/jobs.js]: aw man - that's not what we want, is t?

that'll only work on my machine.

File: src/server/controllers/jobs.js (hunk 4 above)
Comment (jakebrown58) [inline src/server/controllers/jobs.js]: fair point...  i think i'll leave it though since it's consistent with the rest of the function.

**Comment by jakebrown58** [comment]
Made changes based on your comments.

**Pull request merged**